pkgs/development/web/nodejs/v4.nix

```diff
@@ -1,67 +1,20 @@
 { stdenv, fetchurl, openssl, python, zlib, libuv, v8, utillinux, http-parser
 , pkgconfig, runCommand, which, libtool
-}:
-
-# nodejs 0.12 can't be built on armv5tel. Armv6 with FPU, minimum I think.
-# Related post: http://zo0ok.com/techfindings/archives/1820
-assert stdenv.system != "armv5tel-linux";
+, callPackage
+}@args:
 
 let
   version = "4.3.1";
-
-  deps = {
-    inherit openssl zlib libuv;
-
-    # disabled system v8 because v8 3.14 no longer receives security fixes
-    # we fall back to nodejs' internal v8 copy which receives backports for now
-    # inherit v8
-  } // (stdenv.lib.optionalAttrs (!stdenv.isDarwin) {
-    inherit http-parser;
-  });
-
-  sharedConfigureFlags = name: [
-    "--shared-${name}"
-    "--shared-${name}-includes=${builtins.getAttr name deps}/include"
-    "--shared-${name}-libpath=${builtins.getAttr name deps}/lib"
-  ];
-
-  inherit (stdenv.lib) concatMap optional optionals maintainers licenses platforms;
-in stdenv.mkDerivation {
-  name = "nodejs-${version}";
-
   src = fetchurl {
     url = "http://nodejs.org/dist/v${version}/node-v${version}.tar.gz";
     sha256 = "0wzf5sirbph5kaik3pm9i2dxbjwqh5qlnqn71azrsv0vhs7dbqk1";
   };
+in
+  callPackage ./nodejs.nix { 
+    inherit args; 
+    inherit version;
+    inherit src;
 
```

Comment:
IMO no need to define a let-block first. Simply do:

callPackage ./nodejs.nix {
  src = fetchurl { ... };
  version = "4.3.1";
  inherit args;
}

Let block is useful for defining private identifiers whose values need to be reused, but that's not the case here :)